src/cca/ui_components/server_list.ts

```diff
@@ -70,6 +70,8 @@ export class ServerListPage {
       console.debug('Pressed Add Button');
       this.pressAddServer();
     };
+
+    this.servers.onServer(this.addServer).restore();
```

Comment:
Why do you need a listener for a new server? Is there another way to add a server?

I'd suggest this constructor get a list of servers from `servers` and add them to the UI here.